core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostEstimator.scala

```diff
@@ -73,14 +85,21 @@ class DefaultCostEstimator(
     case Some(_) =>
   }
 
-  override def estimate(region: Region, resourceUnits: Int): Double = {
-    this.operatorEstimatedTimeOption match {
+  override def allocateResourcesAndEstimateCost(
+      region: Region,
+      resourceUnits: Int
+  ): (Region, Double) = {
+    // Currently the dummy cost from resourceAllocator is discarded.
+    val (newRegion, _) = resourceAllocator.allocate(region)
+    // We use a cost model that does not rely on the resource allocation.
+    // TODO: Once the ResourceAllocator actually calculates a cost, we can use its calculated cost.
```

Comment:
if we use the calculated cost from the resource allocator, we don't need cost estimator any more. Maybe you should change the name to `region cost` and `region plan cost`, as well as `regionResourceAllocator` and `regionPlanCostEstimator` to differentiate their roles.

Reply:
it is not clear to me how are we going to merge them in the future. if resource allocator will in the future provide an estimated cost, why do we need to wrap a cost estimator outside of it?

Reply:
Ditto